probe-rs/src/core.rs

```diff
@@ -218,9 +219,80 @@ impl<'probe> MemoryInterface for Core<'probe> {
     }
 }
 
+/// A struct containing key information about an exception.
+/// The exception details are architecture specific, and the abstraction is handled in the
+/// architecture specific implementations of [`crate::core::ExceptionInterface`].
+pub struct ExceptionInfo {
+    /// A human readable explanation for the exception.
+    pub description: String,
+    /// The stackframe registers, and their values, for the frame that triggered the exception.
+    pub calling_frame_registers: DebugRegisters,
+}
+
+/// A generic interface to identify and decode exceptions during unwind processing.
+pub trait ExceptionInterface {
+    /// Using the `stackframe_registers` for a "called frame",
+    /// determine if the given frame was called from an exception handler,
+    /// and resolve the relevant details about the exception, including the reason for the exception,
+    /// and the stackframe registers for the frame that triggered the exception.
+    /// A return value of `Ok(None)` indicates that the given frame was called from within the current thread,
+    /// and the unwind should continue normally.
+    fn get_exception_info(
+        &mut self,
+        _stackframe_registers: &DebugRegisters,
+    ) -> Result<Option<ExceptionInfo>, Error> {
+        // For architectures where the exception handling has not been implemented in probe-rs,
+        // this will result in maintaining the current `unwind` behavior, i.e. unwinding will stop
+        // when the first frame is reached that was called from an exception handler.
+        Ok(None)
```

Comment:
Do we really need to return an option if it is not implemen ted? can we not return a not implemented error?

Reply:
Yes, it allows the unwind capabilities for unimplemented architectures to continue working as before.

Reply:
Yes, but what keeps us from matching on the returned Error in the Result for NotImplemented and then not wrap an extra Option? That would be preferred.

Reply:
Done :)